sys/can/conn_can.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (C) 2016 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     can
+ * @{
+ *
+ * @file
+ * @brief       Implementation of generic can connection.
+ *              This file represents a generic can interface on top of Raw can,
+ *              ISOTP, TP 2.0 or BCM layers.
+ *
+ * @author      Aurelien Gonce <aurelien.gonce@altran.com>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <errno.h>
+
+#include "can/conn_can.h"
+#include "can/can.h"
+#include "can/raw.h"
+#include "thread.h"
+#ifdef MODULE_XTIMER
+#include "xtimer.h"
+#endif
+
+#define ENABLE_DEBUG (0)
+#include <debug.h>
+
+#define CONN_CAN_TIMEOUT        (3)
+#define _TIMEOUT_MSG_TYPE       (0x8000)
+#define _TIMEOUT_MSG_VALUE      (0xABCDEFAB)
+
+int conn_can_create(conn_can_t *conn, const void *info, size_t info_len,
+                    int ifnum, can_family_t family, int flags)
+{
+    if (conn == NULL) {
+        DEBUG("conn_can_create: conn object is NULL\n");
+        return -EINVAL;
+    }
+
+    int ret = 0;
+    kernel_pid_t my_pid = thread_getpid();
+
+    /* initialize conn structure */
+    conn->pid = my_pid;
```

Comment:
why not assigning directly from `thread_getpid()`? 'my_pid` is not used anymore within the method.